Firmware/ultralcd.cpp

```diff
@@ -2234,6 +2234,7 @@ void lcd_load_filament_color_check()
 {
 	uint8_t clean = lcd_show_fullscreen_message_yes_no_and_wait_P(_T(MSG_FILAMENT_CLEAN), false, LCD_LEFT_BUTTON_CHOICE);
 	while (clean == LCD_MIDDLE_BUTTON_CHOICE) {
+		lcd_update(2);
```

Comment:
I suspect this will break the MMU menu. I can test it on Friday/Saturday latest.

Reply:
Regarding MMU I can test it or @3d-gussner if he has time. 

lcd_update() should be used with caution because it can create a recursion scenario as in PFW-1531. When lcd_update() is called it forces a re-entry (recursively) into whatever the current LCD menu is rendering. In some situations as PFW-1531 was supposed to fix it can make it seem like the fullscreen buttons (yes/no) dont do anything. When actually the firmware is stuck in menu as it keeps re-entering it. Eventually it will crash the firmware when the stack is full.

Reply:
Thanks for the details!

I went through the code again and I believe forcing the re-entry is what should happen in this particular situation.

Reply:
Yes, works fine 👍 And I also think this more general approach is the much better solution.

Strange thing is, that I'm sure I tried `lcd_draw_update = 2` inside the `while` and I had a much longer delay in screen update. But I'm unable to reproduce this. 🤷